app/[hackathon]/InteractionClient.tsx

```
@@ -0,0 +1,795 @@
+  const [documentation, setDocumentation] = useState("")
+  
+  const searchParams = useSearchParams()
+  const hackAddr = searchParams.get('hackAddr')
+  const urlChainId = searchParams.get('chainId')
+  
+  const chainId = useChainId()
+  const { address: userAddress, isConnected } = useAccount()
+  const { writeContract } = useWriteContract()
+  
+  // Validate hackAddr format
+  const contractAddress = hackAddr && hackAddr.match(/^0x[a-fA-F0-9]{40}$/) 
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)